core/src/cpu/event.rs

```diff
@@ -1,7 +1,64 @@
+use std::array;
+
 use serde::{Deserialize, Serialize};
 
 use crate::runtime::Instruction;
+use crate::runtime::LookupIdSampler;
 use crate::runtime::MemoryRecordEnum;
+use crate::runtime::Opcode;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum LookupIds {
+    AluLookupId(u128),
+    SyscallLookupId(u128),
+    MemoryLookupIds([u128; 2]),
+    BranchLookupIds([u128; 3]),
+    JumpLookupIds([u128; 2]),
+    AuipcLookupId(u128),
+
```

Comment:
nit: this space / comment here is kind of random, would just remove for style reasons